src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_NullableContext.cs

```diff
@@ -0,0 +1,318 @@
+            var expected =
+@"";
+            AssertNullableAttributes(comp, expected);
+        }
+
+        [Fact]
+        public void ExplicitAttribute_FromSource()
+        {
+            var source =
+@"#nullable enable
+public class Program
+{
+    public object F(object arg) => arg;
+}";
+            var comp = CreateCompilation(new[] { NullableContextAttributeDefinition, source });
+            var expected =
+@"Program
+    [NullableContext(1)] System.Object! F(System.Object! arg)
+        System.Object! arg
+";
+            AssertNullableAttributes(comp, expected);
+        }
+
+        [Fact]
+        public void ExplicitAttribute_FromMetadata()
+        {
+            var comp = CreateCompilation(NullableContextAttributeDefinition);
+            comp.VerifyDiagnostics();
+            var ref0 = comp.EmitToImageReference();
+
+            var source =
+@"#nullable enable
+public class Program
+{
+    public object F(object arg) => arg;
+}";
+            comp = CreateCompilation(source, references: new[] { ref0 });
+            var expected =
+@"Program
+    [NullableContext(1)] System.Object! F(System.Object! arg)
+        System.Object! arg
+";
+            AssertNullableAttributes(comp, expected);
+        }
+
+        [Fact]
+        public void ExplicitAttribute_MissingSingleByteConstructor()
+        {
+            var source1 =
+@"namespace System.Runtime.CompilerServices
+{
+    public sealed class NullableContextAttribute : Attribute
+    {
+    }
+}";
+            var source2 =
+@"public class Program
+{
+    public object F(object arg) => arg;
+}";
+
+            // C#7
+            var comp = CreateCompilation(new[] { source1, source2 }, parseOptions: TestOptions.Regular7);
+            comp.VerifyEmitDiagnostics();
+
+            // C#8, nullable disabled
+            comp = CreateCompilation(new[] { source1, source2 }, options: WithNonNullTypesFalse());
+            comp.VerifyEmitDiagnostics();
+
+            // C#8, nullable enabled
+            comp = CreateCompilation(new[] { source1, source2 }, options: WithNonNullTypesTrue());
+            comp.VerifyEmitDiagnostics(
+                // (3,19): error CS0656: Missing compiler required member 'System.Runtime.CompilerServices.NullableContextAttribute..ctor'
+                //     public object F(object arg) => arg;
+                Diagnostic(ErrorCode.ERR_MissingPredefinedMember, "F").WithArguments("System.Runtime.CompilerServices.NullableContextAttribute", ".ctor").WithLocation(3, 19));
+        }
+
+        [Fact]
+        public void ExplicitAttribute_ReferencedInSource()
+        {
+            var sourceAttribute =
+@"namespace System.Runtime.CompilerServices
+{
+    internal class NullableContextAttribute : System.Attribute
+    {
+        internal NullableContextAttribute(byte b) { }
+    }
+}";
+            var source =
+@"#pragma warning disable 169
+using System.Runtime.CompilerServices;
+[assembly: NullableContext(0)]
```

Comment:
Should we get an error for this attribute?

Reply:
Only if `AttributeTargets` is set.

---
In reply to: [296932110](https://github.com/dotnet/roslyn/pull/36152#discussion_r296932110) [](ancestors = 296932110)